src/react/context/index.tsx

```diff
@@ -733,6 +740,37 @@ export const EscolaLMSContextProvider: FunctionComponent<EscolaLMSContextProvide
     [setWebinars],
   );
 
+  const fetchEvents = useCallback(
+    (filter: API.EventsParams) => {
+      setEvents((prevState) => ({ ...prevState, loading: true }));
+      return getEvents(filter)
+        .then((response) => {
+          if (response.success) {
+            setEvents({
+              loading: false,
+              list: response,
+              error: undefined,
+            });
+          }
+          if (response.success === false) {
+            setEvents((prevState) => ({
+              ...prevState,
+              loading: false,
+              error: response,
+            }));
+          }
+        })
+        .catch((error) => {
+          setEvents((prevState) => ({
+            ...prevState,
+            loading: false,
+            error: error,
+          }));
+        });
```

Comment:
this is copy/paste from other parts 
change this to either  
```ts
const fetchStateData<T1, T2> = (fetchAction: Promise<T1>, setState:SetStateAction<T2>):Promise<T1> => {

      setState((prevState) => ({ ...prevState, loading: true }));
      return fetchAction
        .then((response) => {
          if (response.success) {
            setState({
              loading: false,
              list: response,
              error: undefined,
            });
          }
          if (response.success === false) {
            setState((prevState) => ({
              ...prevState,
              loading: false,
              error: response,
            }));
          }
        })
        .catch((error) => {
          setState((prevState) => ({
            ...prevState,
            loading: false,
            error: error,
          }));
        });
} 
```
or custom hook 

then use it something like 

```ts
  const fetchEvents = useCallback(
    (filter: API.EventsParams) => fetchStateData(setEvents, getEvents(filter))
```